sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -2701,8 +2702,11 @@ case class SubtractDates(
 
   override def inputTypes: Seq[AbstractDataType] = Seq(DateType, DateType)
   override def dataType: DataType = {
-    // TODO(SPARK-35727): Return INTERVAL DAY from dates subtraction
-    if (legacyInterval) CalendarIntervalType else DayTimeIntervalType()
+    if (legacyInterval) {
+      CalendarIntervalType
+    } else {
+      DayTimeIntervalType(DAY)
+    }
```

Comment:
nit: minimal changes would be:
```
    if (legacyInterval) CalendarIntervalType else DayTimeIntervalType(DAY)
```

Reply:
@MaxGekk Thank you very much, I fix it. Thank you.